### ISSUE TITLE
`dat share` always gives `(empty archive)`

### ISSUE THREAD

**Issue by cadorn** [datproject/dat#797]
* OS: Darwin Kernel Version 15.5.0
  * node: v5.12.0
  * npm: 3.10.10
  * dat: 13.6.0

When I run:
```
mkdir test
cd test
echo "Hello World" > hello.txt
dat share
```

I always get: `Sharing dat: (empty archive)`

When I run it with `DEBUG=dat-node dat share` I can see `dat-node IMPORT ignore: /hello.txt`.

Why is it ignoring my file? Does not seem to matter what the file is called.

**Comment by mafintosh** [comment]
I can't seem to reproduce this. Are you sure there isn't an old dat in this folder or something?

For the record mine prints

```
dat v13.6.0
<link here>
Sharing dat: 1 files (12 B)

0 connections | Download 0 B/s Upload 0 B/s

Watching for file updates


Ctrl+C to Exit
```

**Comment by cadorn** [comment]
Yeah. Fresh directory. Tried again with node `v6.11.0`. Same issue.

How would we go about troubleshooting this? I am eager to use `dat` as a storage layer in my virtual workspace platform.

Here is my `dat` dependency tree:

```
└─┬ dat@13.6.0
  ├─┬ chalk@1.1.3
  │ ├── ansi-styles@2.2.1
  │ ├── escape-string-regexp@1.0.5
  │ ├─┬ has-ansi@2.0.0
  │ │ └── ansi-regex@2.1.1
  │ ├── strip-ansi@3.0.1
  │ └── supports-color@2.0.0
  ├─┬ cli-truncate@1.0.0
  │ ├── slice-ansi@0.0.4
  │ └─┬ string-width@2.0.0
  │   └── is-fullwidth-code-point@2.0.0
  ├─┬ dat-doctor@1.2.5
  │ ├── crypto@0.0.3
  │ ├── datland-swarm-defaults@1.0.2
  │ ├─┬ discovery-swarm@4.3.8
  │ │ ├── buffer-equals@1.0.4
  │ │ ├── connections@1.4.2
  │ │ ├─┬ discovery-channel@5.4.4
  │ │ │ ├─┬ bittorrent-dht@7.5.3
  │ │ │ │ ├── bencode@0.11.0
  │ │ │ │ ├─┬ k-bucket@3.2.1
  │ │ │ │ │ └── randombytes@2.0.4
  │ │ │ │ ├─┬ k-rpc@4.0.2
  │ │ │ │ │ └── k-rpc-socket@1.6.2
  │ │ │ │ └── lru@3.1.0
  │ │ │ ├── pretty-hash@1.0.1
  │ │ │ └── thunky@0.1.0
  │ │ ├─┬ length-prefixed-message@3.0.3
  │ │ │ └── varint@3.0.1
  │ │ ├── to-buffer@1.1.0
  │ │ └─┬ utp-native@1.5.1
  │ │   ├── nan@2.6.2
  │ │   ├── node-gyp-build@3.2.0
  │ │   └─┬ readable-stream@2.2.11
  │ │     ├── core-util-is@1.0.2
  │ │     ├── isarray@1.0.0
  │ │     ├── safe-buffer@5.0.1
  │ │     ├─┬ string_decoder@1.0.2
  │ │     │ └── safe-buffer@5.0.1
  │ │     └── util-deprecate@1.0.2
  │ ├─┬ dns-discovery@5.6.0
  │ │ ├─┬ dns-socket@1.6.1
  │ │ │ └─┬ dns-packet@1.1.1
  │ │ │   └── ip@1.1.5
  │ │ ├─┬ dns-txt@2.0.2
  │ │ │ └── buffer-indexof@1.1.0
  │ │ ├── lru@2.0.1
  │ │ ├─┬ multicast-dns@6.1.1
  │ │ │ └── thunky@0.1.0
  │ │ ├── network-address@1.1.2
  │ │ └── unordered-set@1.1.0
  │ ├── minimist@1.2.0
  │ ├─┬ pump@1.0.2
  │ │ ├── end-of-stream@1.4.0
  │ │ └─┬ once@1.4.0
  │ │   └── wrappy@1.0.2
  │ └── thunky@1.0.2
  ├─┬ dat-encoding@4.0.2
  │ └── safe-buffer@5.1.0
  ├─┬ dat-json@1.0.0
  │ └─┬ toiletdb@1.4.0
  │   └── last-one-wins@1.0.4
  ├─┬ dat-link-resolve@1.0.0
  │ └─┬ nets@3.2.0
  │   ├─┬ request@2.81.0
  │   │ ├── aws-sign2@0.6.0
  │   │ ├── aws4@1.6.0
  │   │ ├── caseless@0.12.0
  │   │ ├─┬ combined-stream@1.0.5
  │   │ │ └── delayed-stream@1.0.0
  │   │ ├── extend@3.0.1
  │   │ ├── forever-agent@0.6.1
  │   │ ├─┬ form-data@2.1.4
  │   │ │ └── asynckit@0.4.0
  │   │ ├─┬ har-validator@4.2.1
  │   │ │ ├─┬ ajv@4.11.8
  │   │ │ │ ├── co@4.6.0
  │   │ │ │ └─┬ json-stable-stringify@1.0.1
  │   │ │ │   └── jsonify@0.0.0
  │   │ │ └── har-schema@1.0.5
  │   │ ├─┬ hawk@3.1.3
  │   │ │ ├── boom@2.10.1
  │   │ │ ├── cryptiles@2.0.5
  │   │ │ ├── hoek@2.16.3
  │   │ │ └── sntp@1.0.9
  │   │ ├─┬ http-signature@1.1.1
  │   │ │ ├── assert-plus@0.2.0
  │   │ │ ├─┬ jsprim@1.4.0
  │   │ │ │ ├── assert-plus@1.0.0
  │   │ │ │ ├── extsprintf@1.0.2
  │   │ │ │ ├── json-schema@0.2.3
  │   │ │ │ └── verror@1.3.6
  │   │ │ └─┬ sshpk@1.13.0
  │   │ │   ├── asn1@0.2.3
  │   │ │   ├── assert-plus@1.0.0
  │   │ │   ├── bcrypt-pbkdf@1.0.1
  │   │ │   ├─┬ dashdash@1.14.1
  │   │ │   │ └── assert-plus@1.0.0
  │   │ │   ├── ecc-jsbn@0.1.1
  │   │ │   ├─┬ getpass@0.1.7
  │   │ │   │ └── assert-plus@1.0.0
  │   │ │   ├── jodid25519@1.0.2
  │   │ │   └── jsbn@0.1.1
  │   │ ├── is-typedarray@1.0.0
  │   │ ├── json-stringify-safe@5.0.1
  │   │ ├─┬ mime-types@2.1.15
  │   │ │ └── mime-db@1.27.0
  │   │ ├── oauth-sign@0.8.2
  │   │ ├── performance-now@0.2.0
  │   │ ├── qs@6.4.0
  │   │ ├── stringstream@0.0.5
  │   │ ├─┬ tough-cookie@2.3.2
  │   │ │ └── punycode@1.4.1
  │   │ ├── tunnel-agent@0.6.0
  │   │ └── uuid@3.0.1
  │   └─┬ xhr@2.4.0
  │     ├─┬ global@4.3.2
  │     │ ├─┬ min-document@2.19.0
  │     │ │ └── dom-walk@0.1.1
  │     │ └── process@0.5.2
  │     ├── is-function@1.0.1
  │     └─┬ parse-headers@2.0.1
  │       ├── for-each@0.3.2
  │       └── trim@0.0.1
  ├─┬ dat-log@1.1.1
  │ └─┬ random-access-memory@2.4.0
  │   └── process-nextick-args@1.0.7
  ├─┬ dat-node@3.3.2
  │ ├─┬ dat-ignore@1.0.0
  │ │ └─┬ anymatch@1.3.0
  │ │   ├── arrify@1.0.1
  │ │   └─┬ micromatch@2.3.11
  │ │     ├─┬ arr-diff@2.0.0
  │ │     │ └── arr-flatten@1.0.3
  │ │     ├── array-unique@0.2.1
  │ │     ├─┬ braces@1.8.5
  │ │     │ ├─┬ expand-range@1.8.2
  │ │     │ │ └─┬ fill-range@2.2.3
  │ │     │ │   ├── isobject@2.1.0
  │ │     │ │   ├── randomatic@1.1.6
  │ │     │ │   └── repeat-string@1.6.1
  │ │     │ ├── preserve@0.2.0
  │ │     │ └── repeat-element@1.1.2
  │ │     ├─┬ expand-brackets@0.1.5
  │ │     │ └── is-posix-bracket@0.1.1
  │ │     ├── extglob@0.3.2
  │ │     ├── filename-regex@2.0.1
  │ │     ├── is-extglob@1.0.0
  │ │     ├── is-glob@2.0.1
  │ │     ├─┬ object.omit@2.0.1
  │ │     │ ├─┬ for-own@0.1.5
  │ │     │ │ └── for-in@1.0.2
  │ │     │ └── is-extendable@0.1.1
  │ │     ├─┬ parse-glob@3.0.4
  │ │     │ ├─┬ glob-base@0.3.0
  │ │     │ │ └── glob-parent@2.0.0
  │ │     │ └── is-dotfile@1.0.3
  │ │     └─┬ regex-cache@0.4.3
  │ │       ├── is-equal-shallow@0.1.3
  │ │       └── is-primitive@2.0.0
  │ ├─┬ dat-storage@1.0.2
  │ │ ├─┬ append-tree@2.3.5
  │ │ │ ├── array-lru@1.1.1
  │ │ │ ├─┬ brfs@1.4.3
  │ │ │ │ ├─┬ quote-stream@1.0.2
  │ │ │ │ │ └── buffer-equal@0.0.1
  │ │ │ │ ├─┬ resolve@1.3.3
  │ │ │ │ │ └── path-parse@1.0.5
  │ │ │ │ └─┬ static-module@1.3.2
  │ │ │ │   ├─┬ concat-stream@1.6.0
  │ │ │ │   │ └── typedarray@0.0.6
  │ │ │ │   ├─┬ duplexer2@0.0.2
  │ │ │ │   │ └─┬ readable-stream@1.1.14
  │ │ │ │   │   ├── isarray@0.0.1
  │ │ │ │   │   └── string_decoder@0.10.31
  │ │ │ │   ├─┬ escodegen@1.3.3
  │ │ │ │   │ ├── esprima@1.1.1
  │ │ │ │   │ ├── estraverse@1.5.1
  │ │ │ │   │ ├── esutils@1.0.0
  │ │ │ │   │ └─┬ source-map@0.1.43
  │ │ │ │   │   └── amdefine@1.0.1
  │ │ │ │   ├─┬ falafel@1.2.0
  │ │ │ │   │ ├── acorn@1.2.2
  │ │ │ │   │ ├── foreach@2.0.5
  │ │ │ │   │ ├── isarray@0.0.1
  │ │ │ │   │ └── object-keys@1.0.11
  │ │ │ │   ├─┬ has@1.0.1
  │ │ │ │   │ └── function-bind@1.1.0
  │ │ │ │   ├── object-inspect@0.4.0
  │ │ │ │   ├─┬ quote-stream@0.0.0
  │ │ │ │   │ └── minimist@0.0.8
  │ │ │ │   ├─┬ readable-stream@1.0.34
  │ │ │ │   │ ├── isarray@0.0.1
  │ │ │ │   │ └── string_decoder@0.10.31
  │ │ │ │   ├── shallow-copy@0.0.1
  │ │ │ │   ├─┬ static-eval@0.2.4
  │ │ │ │   │ └─┬ escodegen@0.0.28
  │ │ │ │   │   ├── esprima@1.0.4
  │ │ │ │   │   └── estraverse@1.3.2
  │ │ │ │   └─┬ through2@0.4.2
  │ │ │ │     └─┬ xtend@2.1.2
  │ │ │ │       └── object-keys@0.4.0
  │ │ │ ├── codecs@1.2.0
  │ │ │ └── varint@5.0.0
  │ │ ├── dat-secret-storage@4.0.0
  │ │ └─┬ multi-random-access@2.1.1
  │ │   ├── abstract-random-access@1.1.2
  │ │   └── sorted-array-functions@1.0.0
  │ ├── dat-swarm-defaults@1.0.0
  │ ├─┬ hyperdrive@9.3.0
  │ │ ├─┬ duplexify@3.5.0
  │ │ │ └─┬ end-of-stream@1.0.0
  │ │ │   └── once@1.3.3
  │ │ ├── from2@2.3.0
  │ │ ├─┬ hypercore@6.5.0
  │ │ │ ├── atomic-batcher@1.0.2
  │ │ │ ├─┬ bitfield-rle@2.1.0
  │ │ │ │ └── varint@4.0.1
  │ │ │ ├── bulk-write-stream@1.1.3
  │ │ │ ├── flat-tree@1.6.0
  │ │ │ ├─┬ hypercore-protocol@6.4.0
  │ │ │ │ ├── sorted-indexof@1.0.0
  │ │ │ │ └── varint@5.0.0
  │ │ │ ├── memory-pager@1.0.3
  │ │ │ ├── merkle-tree-stream@3.0.3
  │ │ │ ├─┬ sodium-signatures@2.0.0
  │ │ │ │ └── tweetnacl@0.14.5
  │ │ │ ├── sparse-bitfield@3.0.3
  │ │ │ ├── unordered-array-remove@1.0.2
  │ │ │ └── unordered-set@2.0.0
  │ │ ├── inherits@2.0.3
  │ │ ├── mutexify@1.1.0
  │ │ ├─┬ protocol-buffers@3.2.1
  │ │ │ ├── generate-function@2.0.0
  │ │ │ ├─┬ generate-object-property@1.2.0
  │ │ │ │ └── is-property@1.0.2
  │ │ │ ├── protocol-buffers-schema@3.3.1
  │ │ │ ├─┬ signed-varint@2.0.1
  │ │ │ │ └── varint@5.0.0
  │ │ │ └── varint@5.0.0
  │ │ ├─┬ sodium-universal@1.0.0
  │ │ │ ├─┬ sodium-javascript@0.0.1
  │ │ │ │ └── blakejs@1.1.0
  │ │ │ └── sodium-native@1.10.1
  │ │ ├── stream-collector@1.0.1
  │ │ ├── uint64be@2.0.1
  │ │ └─┬ unixify@1.0.0
  │ │   └─┬ normalize-path@2.1.1
  │ │     └── remove-trailing-separator@1.0.1
  │ ├── hyperdrive-network-speed@2.0.1
  │ ├─┬ mirror-folder@2.1.1
  │ │ ├── fd-read-stream@1.1.0
  │ │ └─┬ recursive-watch@1.1.2
  │ │   └── ttl@1.3.0
  │ ├── multicb@1.2.1
  │ ├─┬ random-access-file@1.7.3
  │ │ ├── buffer-alloc-unsafe@1.0.0
  │ │ └─┬ mkdirp@0.5.1
  │ │   └── minimist@0.0.8
  │ ├─┬ stream-each@1.2.0
  │ │ └── stream-shift@1.0.0
  │ └── untildify@3.0.2
  ├─┬ dat-registry@3.0.2
  │ └─┬ township-client@1.3.1
  │   ├─┬ is-number@2.1.0
  │   │ └─┬ kind-of@3.2.2
  │   │   └── is-buffer@1.1.5
  │   ├── is-string@1.0.4
  │   └── os-homedir@1.0.2
  ├─┬ debug@2.6.8
  │ └── ms@2.0.0
  ├─┬ hyperdrive-http@4.2.0
  │ ├─┬ corsify@2.1.0
  │ │ └─┬ http-methods@0.1.0
  │ │   ├── body@0.1.0
  │ │   └─┬ content-types@0.1.0
  │ │     └─┬ iterators@0.1.0
  │ │       └── ap@0.1.0
  │ ├── directory-index-html@2.1.0
  │ ├── mime@1.3.6
  │ └── range-parser@1.2.0
  ├─┬ neat-log@1.1.0
  │ ├── lodash.throttle@4.1.1
  │ ├─┬ nanobus@3.3.0
  │ │ └── nanotiming@1.0.1
  │ └─┬ status-logger@3.1.0
  │   └─┬ ansi-diff-stream@1.2.0
  │     └── through2@2.0.3
  ├── prettier-bytes@1.0.4
  ├── progress-string@1.2.1
  ├─┬ prompt@1.0.0
  │ ├── colors@1.1.2
  │ ├── pkginfo@0.4.0
  │ ├─┬ read@1.0.7
  │ │ └── mute-stream@0.0.7
  │ ├── revalidator@0.1.8
  │ ├─┬ utile@0.3.0
  │ │ ├── async@0.9.2
  │ │ ├── deep-equal@0.2.2
  │ │ ├── i@0.3.5
  │ │ └── ncp@1.0.1
  │ └─┬ winston@2.1.1
  │   ├── async@1.0.0
  │   ├── colors@1.0.3
  │   ├── cycle@1.0.3
  │   ├── eyes@0.1.8
  │   ├── isstream@0.1.2
  │   ├── pkginfo@0.3.1
  │   └── stack-trace@0.0.10
  ├─┬ rimraf@2.6.1
  │ └─┬ glob@7.1.2
  │   ├── fs.realpath@1.0.0
  │   ├── inflight@1.0.6
  │   ├─┬ minimatch@3.0.4
  │   │ └─┬ brace-expansion@1.1.7
  │   │   ├── balanced-match@0.4.2
  │   │   └── concat-map@0.0.1
  │   └── path-is-absolute@1.0.1
  ├── speedometer@1.0.0
  ├─┬ subcommand@2.1.0
  │ └── cliclopts@1.1.1
  └── xtend@4.0.1
```

**Comment by mafintosh** [comment]
@joehand any idea why dat-node would be ignoring all files?

**Comment by cadorn** [comment]
I would be happy to do a screen sharing session. I tried debugging it a bit but feel like I am stabbing in the dark as it should obviously be working.

**Comment by joehand** [comment]
@cadorn thanks for the debug info. It seems like it is something about the directory name/structure, instead of the file name.

Few follow up questions:
* Are you sharing from the root as printed in debug? Or another directory? (I followed your instructions creating `/test` and sharing that, it worked ok).
* Does it work if you share a different directory, e.g. `dat share ~/Desktop/test`?

It seems like it is something in the full path we are ignoring by accident.

**Comment by joehand** [comment]
Oh and can jump on a screen share too! Ping me, `jhand`, on our chat [gitter](https://gitter.im/datproject/discussions) or #dat on irc.

**Comment by cadorn** [comment]
> Are you sharing from the root as printed in debug? Or another directory?

Good catch! I had a `.tmp` segment in my parent directory path which IMO should be valid.

Here is a minimal failing test case:
```
cd ~/Desktop

rm -Rf .tmp || true
mkdir .tmp
cd .tmp

nvm use 5
npm install -g dat

echo "Hello World" > hello.txt
dat share
```

**Comment by joehand** [comment]
Published a new version of `dat-ignore@1.1.1`, which should fix this and added a test to catch this in dat-node =).

Thanks for catching this.
